### PR TITLE
rework Button docs to include a Best Practices section

### PR DESCRIPTION
**Pull request by kevinbarabash**
The key changes:
- there's more regularity between the main examples (each focuses on a particular prop)
- the best practices section covers issues that people might run into an common solutions as well as best practices
- I also changed the `size` prop to take `"small" | "medium" (default)` instead of `"small" | "default" (default)` so that it would mesh better with our other `size` props.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit e4046dd33e1373e7192bf47f79061ad9045ede35

https://deploy-preview-253--wonder-blocks.netlify.com

File: packages/wonder-blocks-button/components/button.md [hunk 1]
```diff
@@ -93,171 +141,280 @@ const styles = StyleSheet.create({
...
+        </Button>
+    </View>
+</View>
+```
+
+Layouts often specify a specific width of button.  When implementing such
+designs use `minWidth` instead of `width`.  It prevents a button from
+resizing to fit content.  This is important for international sites since
+sometimes strings for UI elements can be much longer than their English
+counterparts.
```
Comment (chenyejia) [inline packages/wonder-blocks-button/components/button.md]: I'm confused by this part. Is "It prevents a button..." referring to `minWidth` or `width`? Is `badButton` supposed to be with `minWidth`?

File: packages/wonder-blocks-button/components/button.md [hunk 2]
```diff
@@ -1,78 +1,126 @@
...
 </View>
 ```
 
-Button example, `href="#button-1"`:
-```js
-<Button
-    onClick={(e) => console.log("Hello, world!")}
-    href="#button-1"
->Label</Button>
+Buttons have a `color` that is either `"default"` (default) or `"destructive"`:
```
Comment (jeresig) [inline packages/wonder-blocks-button/components/button.md]: Might be clearer to say something like `(the default, as shown above) or "destructive" (as can be seen below):`

File: packages/wonder-blocks-button/components/button.md [hunk 3]
```diff
@@ -83,181 +131,325 @@ const styles = StyleSheet.create({
...
-    >Disabled</Button>
+        disabled={true}
+    >
+        Tertiary
+    </Button>
 </View>
 ```
 
-Button examples, `style` specified (Supports width, position, margin, and flex styles):
+Buttons have a `size` that's either `"medium"` (default) or `"small"`.
```
Comment (jeresig) [inline packages/wonder-blocks-button/components/button.md]: Would be good to clarify that the following buttons are all "small".

File: packages/wonder-blocks-button/components/button.md [hunk 4]
```diff
@@ -83,181 +131,325 @@ const styles = StyleSheet.create({
...
+<View style={styles.row}>
+    <Button
+        href="#button-1"
+        style={styles.button}
+    >
+        href
+    </Button>
+    <Button
+        kind="secondary"
+        onClick={(e) => console.log("Hello, world!")}
```
Comment (jeresig) [inline packages/wonder-blocks-button/components/button.md]: When I clicked these I was kind of surprised when something didn't happen. Maybe an `alert` might be more obvious?

File: packages/wonder-blocks-button/components/button.md [hunk 5]
```diff
@@ -1,78 +1,126 @@
...
-    >Button with href</Button>
+    >
+        Tertiary
+    </Button>
 </View>
 ```
 
-Buttons on a dark blue background:
-```js
+Buttons on a `darkBlue` background should be `light`.
```
Comment (jeresig) [inline packages/wonder-blocks-button/components/button.md]: Maybe saying "should set `light` to `true`" would make it clearer.

File: packages/wonder-blocks-button/components/button.md [hunk 6]
```diff
@@ -83,181 +131,325 @@ const styles = StyleSheet.create({
...
+            </Route>
+        </Switch>
+    </View>
+</MemoryRouter>
+```
+
+Buttons can have a `style` props which supports width, position, margin,
+and flex styles:
+
+TODO(kevinb): link to front matter section describing the `style` prop in
```
Comment (jeresig) [inline packages/wonder-blocks-button/components/button.md]: I wish there was a way to include non-visible comments. It makes for a very awkward documentation reading experience. I think it'd be better to put these into Jira issues instead.

File: packages/wonder-blocks-button/components/button.md [hunk 7]
```diff
@@ -83,181 +131,325 @@ const styles = StyleSheet.create({
...
+            label
+        </Button>
+        <Button
+            style={styles.goodButton}
+        >
+            label in a different language
+        </Button>
+    </View>
+    <View style={styles.gap} />
+    <View style={styles.row}>
```
Comment (jeresig) [inline packages/wonder-blocks-button/components/button.md]: I think it'd be better to just remove the bad example from this as it makes it confusing as to which is the right one (or we should add text explaining which set of buttons are using which technique).

File: packages/wonder-blocks-button/components/button.md [hunk 8]
```diff
@@ -83,181 +131,325 @@ const styles = StyleSheet.create({
...
+        <Button
+            style={styles.badButton}
+        >
+            Primary
+        </Button>
+    </View>
+</View>
 ```
+
+TODO(kevinb): add section on media queries.
```
Comment (jeresig) [inline packages/wonder-blocks-button/components/button.md]: (Same goes for all of the in-docs TODOs)

File: packages/wonder-blocks-button/components/button.md (hunk 3 above)
Comment (kevinbarabash) [reply packages/wonder-blocks-button/components/button.md]: Will do.  I should note that this is a breaking change.  Should I bump button to `v2`?

Comment (jeresig) [reply packages/wonder-blocks-button/components/button.md]: Yep, I think so!

**Comment by kevinbarabash** [comment]
@jeresig could you have another look?

**Pull request merged**